### PR TITLE
KCL-10481 - Properly track all necessary mutations when resolving com…

### PR DESCRIPTION
**Pull request by hejtmii**

### CODE REVIEW

File: src/component/selection/getOffsetKeyFromNode.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
```
Comment (vilemj-Viclick) [inline src/component/selection/getOffsetKeyFromNode.js]: Is it though?

I mean... This is not really a Facebook file...

File: src/component/handlers/composition/DraftEditorCompositionHandler.js [hunk 2]
```diff
@@ -178,7 +186,17 @@ const DraftEditorCompositionHandler = {
     // }
 
     let contentState = editorState.getCurrentContent();
-    mutations.forEach((composedChars, offsetKey) => {
+    let mutatedEditorState = editorState;
+
+    const updateEditorState = () => {
+      // We need to update the editorState so the leaf node ranges are properly
+      // updated and multiple mutations are correctly applied.
+      mutatedEditorState = EditorState.set(mutatedEditorState, {
```
Comment (vilemj-Viclick) [inline src/component/handlers/composition/DraftEditorCompositionHandler.js:194]: You may want to explain why `set` and not `push`. Or in general why this unorthodox method?

**Review by vilemj-Viclick — CHANGES_REQUESTED** [review]

File: src/component/handlers/composition/DraftEditorCompositionHandler.js (hunk 2 above)
Comment (hejtmii) [reply src/component/handlers/composition/DraftEditorCompositionHandler.js:194]: This comes from the original code, I just moved it higher. `set` sets the properties, while `push` updates content, and (optionally) adds item to undo stack. They have quite different contract also

File: src/component/selection/getOffsetKeyFromNode.js (hunk 1 above)
Comment (hejtmii) [reply src/component/selection/getOffsetKeyFromNode.js]: Technically no, but I would like to create PR to original repo from this and every file has this regardless of the author :-)

**Review by petersolar — APPROVED** [review]

**Comment by hejtmii** [comment]
Abandoning in favor of https://github.com/kontent-ai/draft-js/pull/17 which should be more reliable

**Pull request closed without merging**